File: ci/openmp_test.f90
program openmp_test
    use omp_lib
    implicit none
    integer :: tid, nthreads
    
    !$omp parallel private(tid, nthreads)
    tid = omp_get_thread_num()
    nthreads = omp_get_num_threads()
    !$omp critical
    print *, 'Hello from OpenMP thread', tid, 'of', nthreads
    !$omp end critical
    !$omp end parallel
end program openmp_test

File: ci/openmp_test.c
#include <stdio.h>
#include <omp.h>

int main() {
    #pragma omp parallel
    {
        int tid = omp_get_thread_num();
        int nthreads = omp_get_num_threads();
        #pragma omp critical
        printf("Hello from OpenMP thread %d of %d\n", tid, nthreads);
    }
    return 0;
}
